### ISSUE TITLE
Issues when oembed.com/providers.json has issues

### ISSUE THREAD

**Issue by jpadilla** [pyembed/pyembed#60]
http://oembed.com/providers.json is returning `false` probably related to https://github.com/iamcal/oembed/pull/112.

`_build_endpoints_from_providers` always expects something that can be iterated on, this fails since `providers` in this case is `False`. Two ways to fix this on `UrlDiscoverer`: Make sure we raise if response.json() is `False` or make sure we're always sending a list to `_build_endpoints_from_providers`.

This also happens on importing pyembed since discoverer is executed on `PyEmbed.__init__()` which is on the module's core `__init__` so importing anything else to work around this is next to imposible. How about instead doing something like:

```python
class PyEmbed(object):
    def __init__(self, discoverer=DefaultDiscoverer, renderer=DefaultRenderer):
        """OEmbed consumer with automatic discovery.
        :param renderer: (optional) renderer to render the response.
        """
        self.discoverer = discoverer()
        self.renderer = renderer()
```

Thoughts? I can put something together.

**Comment by matt-thomson** [comment]
Which version are you using? There was an issue around hitting http://oembed.com/providers.json on import, but this was fixed in v1.2.3.

Thanks for your PR - I'll take a look at that now.

**Comment by jpadilla** [comment]
Resolved by #61
